networkx/generators/degree_seq.py

```diff
@@ -31,8 +38,123 @@
            'degree_sequence_tree',
            'random_degree_sequence_graph']
 
+chaini = chain.from_iterable
+
+
+# TODO Can there be nonnegative integers in the degree sequence? If so,
+# change the documentation of this method (and the other configuration
+# model functions) to indicate that, and give an additional example.
+def _to_stublist(degree_sequence):
+    """Returns a list of degree-repeated node numbers.
+
+    ``degree_sequence`` is a list of positive integers representing the
+    degrees of nodes in a graph.
+
+    This function returns a list of node numbers with multiplicities
+    according to the given degree sequence. For example, if the first
+    element of ``degree_sequence`` is ``3``, then the first node number,
+    ``0``, will appear at the head of the returned list three times. The
+    node numbers are assumed to be the numbers zero through
+    ``len(degree_sequence) - 1``.
+
+    For example::
+
+        >>> degree_sequence = [1, 2, 3]
+        >>> _to_stublist(degree_sequence)
+        [0, 1, 1, 2, 2, 2]
+
+    """
+    return list(chaini([n] * d for n, d in enumerate(degree_sequence)))
+
+
+def _random_balanced_bipartition(sequence):
```

Comment:
Sure. I think I did this because I wasn't sure if this was the best implementation, so I wanted it to be easily replaceable.